### PR TITLE
WebP image support

### PR DESCRIPTION
**Pull request by jroebu14**
Part of https://github.com/bbc/simorgh/issues/9628

**Overall change:**
Adds WebP support across areas that we use the `/news/` iChef location. This is achieved by having a `srcset` and `fallbackSrcset` for images to render WebP images if supported, or fallback to the `fallbackSrcset` of standard JPG/PNG.

There are a couple of components that are not using WebP at the moment:
- Components that use `placeholderSrcset`
- Inline podcast promos
- Includes
- OnDemandImage

These components don't consistently use the `https://ichef.bbci.co.uk/news/...` location. **Only** the `/news/` iChef location from `cpsprodpb` and `cpsdevpb` origins supports WebP currently, so to prevent unnecessary 404s, these components don't make use of the additional `fallbackSrcset` logic, keeping their primary `srcset` to JPG/PNG as the do now. 

**Code changes:**

- Updates usage of the `Image` to pass through `srcset` and `fallbackSrcset` strings
- Updates the `createSrcset` function to return an object that contains WebP URLs and standard JPG/PNG URLs
- Updates statically set iChef URLs to make use of the existing `buildIChefURL` function to ensure the correct iChef location is used with the correct file extension. The `createSrcsets` function also uses this to constructor WebP srcsets

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/lib/utilities/srcSet/index.js [hunk 1]
```diff
@@ -2,39 +2,63 @@ import buildIChefUrl from '#lib/utilities/ichefURL';
...
+    .map(
+      resolution =>
+        `${buildIChefUrl({
+          originCode,
+          locator,
+          resolution,
+          isWebP: false,
+        })} ${resolution}w`,
+    )
+    .join(', ');
```
Comment (jroebu14) [inline src/app/lib/utilities/srcSet/index.js]: Could maybe be shortened to:

```suggestion
    const [webpSrcset, fallbackSrcset] = [true, false].map(isWebP =>
    requiredResolutions
        .map(
        resolution =>
            `${buildIChefUrl({
            originCode,
            locator,
            resolution,
            isWebP,
            })} ${resolution}w`,
        )
        .join(', '),
    )

```

I haven't tested this code though

**Review by jroebu14 — COMMENTED** [review]
Looks really good 👍

File: src/app/lib/utilities/srcSet/index.js (hunk 1 above)
Comment (amoore108) [reply src/app/lib/utilities/srcSet/index.js]: Nice, works well 👍🏼

File: src/app/lib/utilities/srcSet/index.js [hunk 2]
```diff
@@ -23,30 +23,20 @@ export const createSrcsets = ({
...
+          `${buildIChefUrl({
+            originCode,
+            locator,
+            resolution,
+            isWebP,
+          })} ${resolution}w`,
+      )
+      .join(', '),
+  );
+  console.log({ webpSrcset, fallbackSrcset });
```
Comment (jroebu14) [inline src/app/lib/utilities/srcSet/index.js]: sneaky console log

**Review by amoore108 — APPROVED** [review]

File: src/app/lib/utilities/ichefURL/index.js [hunk 3]
```diff
@@ -32,7 +35,7 @@ const buildIChefURL = ({ originCode, locator, resolution, isWebP = false }) => {
     locator,
   ].join('/');
 
-  if (isWebP) {
+  if (isWebP && WEBP_ORIGIN_CODES.includes(originCode)) {
```
Comment (jroebu14) [inline src/app/lib/utilities/ichefURL/index.js]: could be a const named `webpIsSupported` or something

**Review by ryanmccombe — APPROVED** [review]

**Review by rebeccamcginn — APPROVED** [review]

**Pull request merged**